src/loki/reach.cc

```diff
@@ -78,41 +55,36 @@ directed_reach Reach::operator()(const DirectedEdge* edge,
     // expand from the node
     if (!reader.GetGraphTile(node_id, tile))
       continue;
-    for (const auto& edge : tile->GetDirectedEdges(node_id)) {
+    const auto* node = tile->node(node_id);
+
+    auto directed_edges = tile->GetDirectedEdges(node_id);
+    for (const auto* edge = directed_edges.begin(); edge != directed_edges.end(); ++edge) {
       // TODO: we'd rather say !edge.end_simple_restriction() and not !edge.restrictions()
       // TODO: but we'd need the predecessor information to do that so we punt 1 edge earlier
       // NOTE: we can go through the start of the restriction because only the end would mark a
       // potential stopping point (maybe a path followed the restriction)
 
+      GraphId edge_id = tile->id();
+      edge_id.set_id(node->edge_index() + (edge - directed_edges.begin()));
```

Comment:
instead of this, you can make the edge_id outside o the loop, and then `++edge_id` up there where you `++edge`. we have a prefix `++` operator for graph ids that works as you would expect